### PR TITLE
log in rd 2

### PR DESCRIPTION
**Pull request by ridhoq**

### CODE REVIEW

File: test/client-setup.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+var jsdom = require('jsdom').jsdom;
```
Comment (austinpray) [inline test/client-setup.js:1]: I mean you should prolly use karma

Comment (ridhoq) [reply test/client-setup.js:1]: meh enzyme docs recommended jsdom:
http://airbnb.io/enzyme/docs/guides/jsdom.html

Comment (austinpray) [reply test/client-setup.js:1]: run your tests in a real browser reee

File: client/components/login.js [hunk 2]
```diff
@@ -1,33 +1,98 @@
...
+            this.props.handleLogIn(this.state.user);
+        }
+    };
+
+    handleUsernameChange = (e) => {
+        const val = e.target.value;
+        if (!validator.isEmpty(val) && validator.isAscii(val)) {
+            this.setState({
+                validUsername: true,
+                user: update(this.state.user, {username: {$set: val}})
```
Comment (austinpray) [inline client/components/login.js:28]: why are you using update?

Object.assign({}, this.state.user, {username: val});

Comment (ridhoq) [reply client/components/login.js:28]: yeah agreed, that's the better way to do it

Comment (austinpray) [reply client/components/login.js:28]: also you want to keep the state as non-nested as possible. prefer `{username: 'test', password: 'dop'}` over `{user: {username: 'test', password: 'dop'}}`

ints, strings, numbers are natively immutable types in js

**Comment by ridhoq** [comment]
@austinpray I'm gonna go ahead and merge this - if you have any C O M P E L L I N G feedback, I'll address it in a future PR

**Pull request merged**